tests/ferret/core_test.go

```diff
@@ -326,10 +372,15 @@ func TestCore(t *testing.T) {
 				require.NotNil(t, cursor)
 
 				var expected []bson.D
+
 				for _, id := range tc.IDs {
 					v, ok := data[id]
 					require.True(t, ok)
-					expected = append(expected, bson.D{{"_id", id}, {"value", v}})
+					if tc.v != nil {
+						expected = append(expected, tc.v)
```

Comment:
I think it makes it much more complicated than it is worth. Here we iterate over IDs, get the value by it – and then ignore it. Why?

And also why do we do that at all? All those values are available by IDs, that was the point – to avoid copying values.

Reply:
If to copy values only, then cannot check that the projection works.
in `TestCore/QueryOperators/ProjectionExclusion` the value is not included in the result

Reply:
logic fixed :)